### ISSUE TITLE
Remove whitespace at the start of the clang-tidy fix modernize-use-override 

### ISSUE THREAD

**Issue by hriekehof** [clangd/vscode-clangd#496]
If i have a class in a header which inherits from a pure virtual base class i observe the following when i apply the fix.

```c++
class A : public Interface
{
    // This will be marked by clangd to use override instead of virtual
    virtual void Reset();

    // After applying the fix i get this and i always have to manually delete the one whitespace :)
     void Reset() override;
};
````
I am not sure if this problem also occurs for other checks where a fix is applied via `clangd`

**System information**
Clangd version (from the log, or `clangd --version`):
```
Apple clangd version 15.0.0 (clang-1500.0.29.1)
Features: mac+xpc
Platform: x86_64-apple-darwin22.5.0; target=arm64-apple-darwin22.5.0
```
clangd extension version: `0.1.24`
Operating system: `macOS Ventura 13.4.1`

**Comment by HighCommander4** [comment]
Confirmed. Does not occur with `clang-tidy --fix`, so the issue seems to be clangd-specific.

**Comment by HighCommander4** [comment]
I think this would make a good first issue for someone who'd like to get involved in clangd / learn about LSP to investigate.

**Comment by CosmicPegasis** [comment]
Hi, I'm interesting in solving this, however I don't have a lot of prior experience with the LSP. How can I start?

**Comment by HighCommander4** [comment]
Thanks for your interest!

I think some good first investigative steps are:

 1. Verify that you can reproduce the issue locally
 2. Look at clangd logs to determine whether the issue is on the server side or the client side:
    * Add `"--log=verbose"` and `"--pretty"` to `"clangd.arguments"` in vscode settings. (The first causes clangd to include the contents of all LSP traffic in the logs, and the second causes clangd to format the JSON in the LSP messages in a more readable way.)
    * Restart clangd (F1 --> "clangd: Restart language server")
    * Reproduce the issue
    * Open the "Output" view and select "clangd" from the dropdown to look at the logs

In the logs, find the request and response relevant to this issue (`textDocument/codeAction`), and see whether it's the server sending an incorrect response (indicating an issue on the server side), or the client interpreting the response incorrectly (indicating an issue on the client side).

To help understand the intended interpretation of the LSP message contents, you can look at https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/ as a reference.

If you're unsure, feel free to post the logs here and I can help with the analysis.

**Comment by CosmicPegasis** [comment]
Hi, yeah since my previous comment I have been trying to reproduce the issue. However, I'm unable to. I'm guessing I misinterpreted the question.
I setup vscode-clangd.
Then I tried to reproduce the issue by the following
![image](https://github.com/clangd/vscode-clangd/assets/65207788/d817f2cd-f718-4a5e-b8e5-362c171c7cbd)
As you can see I'm not getting any squiggles.

When I copy and paste the code exactly in the original post, i.e. 
![image](https://github.com/clangd/vscode-clangd/assets/65207788/05f1e41a-fe3e-4393-82a1-580f3cd680c1)

Although the code doesn't make sense, I get the yellow error squiggle which says the method should have override.
But I still don't get a quick fix option.

![image](https://github.com/clangd/vscode-clangd/assets/65207788/9e226204-32d0-413b-a29b-20b792952f4a)

**Comment by HighCommander4** [comment]
You need to enable the clang-tidy check which issues this diagnostic.

Try adding a `.clang-tidy` file in the root directory of your project containing:

```
Checks: 'cppcoreguidelines-explicit-virtual-functions'
```

Then you should see the diagnostic issued for code like this:

```c++
class Interface {
  virtual void Reset() = 0;
};
class A : public Interface {
  virtual void Reset();
};
```

If you hover over the diagnostic, you should see:

> Prefer using 'override' or (rarely) 'final' instead of 'virtual' (fix available)clang-tidy[cppcoreguidelines-explicit-virtual-functions](https://clang.llvm.org/extra/clang-tidy/checks/cppcoreguidelines/explicit-virtual-functions.html)

and it should have a quick fix.

**Comment by CosmicPegasis** [comment]
Yep! It seems like the LSP itself just deletes the virtual keyword and not the space after it.
```
I[11:07:30.008] <-- textDocument/codeAction(10)
V[11:07:30.008] ASTWorker running EnumerateTweaks on version 51 of /home/cosmic/coding/cpp/repro/main.hpp
I[11:07:30.008] --> reply:textDocument/codeAction(10) 0 ms
V[11:07:30.008] >>>
 ```
```json
{
  "id": 10,
  "jsonrpc": "2.0",
  "result": [
    {
      "diagnostics": [
        {
          "code": "cppcoreguidelines-explicit-virtual-functions",
          "message": "Prefer using 'override' or (rarely) 'final' instead of 'virtual' (fix available)",
          "range": {
            "end": {
              "character": 22,
              "line": 6
            },
            "start": {
              "character": 17,
              "line": 6
            }
          },
          "severity": 2,
          "source": "clang-tidy"
        }
      ],
      "edit": {
        "changes": {
          "file:///home/cosmic/coding/cpp/repro/main.hpp": [
            {
              "newText": " override",
              "range": {
                "end": {
                  "character": 24,
                  "line": 6
                },
                "start": {
                  "character": 24,
                  "line": 6
                }
              }
            },
            {
              "newText": "",
              "range": {
                "end": {
                  "character": 11,
                  "line": 6
                },
                "start": {
                  "character": 4,
                  "line": 6
                }
              }
            }
          ]
        }
      },
      "isPreferred": true,
      "kind": "quickfix",
      "title": "prefer using 'override' or (rarely) 'final' instead of 'virtual'"
    }
  ]
}
```
The ``end`` specification for ``""`` part i.e. the delete command deletes only the virtual keyword and not the space after it since indexing is zero-based and end exclusive.
It seems like an issue with clangd.

**Comment by HighCommander4** [comment]
Thanks, I agree with your analysis. I'm going to transfer the issue into the server repo now that we now it's a server issue.
